src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingController.java

```diff
@@ -331,6 +335,24 @@ private void selectFactions(List<Faction> factions) {
     factionsToButtons.forEach((faction, toggleButton) -> toggleButton.setSelected(factions.contains(faction)));
   }
 
+  protected void onMapPoolClickedListener(MatchmakerQueueInfo queue) {
+    TeamMatchmakingMapListController controller = uiService.loadFxml("theme/play/teammatchmaking/matchmaking_maplist_popup.fxml");
+    DialogLayout dialogLayout = new DialogLayout();
+    dialogLayout.setHeading(new Label(i18n.get("teammatchmaking.mapPool")));
+    dialogLayout.setBody(controller.getRoot());
+    Dialog dialog = new Dialog();
+    dialog.setContent(dialogLayout);
+    dialog.setTransitionType(DialogTransition.CENTER);
+    teamMatchmakingRoot.setOnKeyPressed(event -> {
+      if (event.getCode() == KeyCode.ESCAPE) {
+        dialog.close();
+      }
+    });
+    controller.init(queue, dialogLayout);
```

Comment:
This is done backwards. The controller should not need to know it is being placed in a dialog.

You should only need to use the getRoot method to get the parent element of the controller and add it to the dialog.

Additionally queue should have its own setter.

Reply:
But i need dialog root in my controller to calculate width/height of the popup reliably
I can do it with cycled Node.getParent as alternative, but it is less reliable because it is dialog-structure-dependent